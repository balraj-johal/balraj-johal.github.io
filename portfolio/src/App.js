import { BrowserRouter as Router, Route, Switch } from "react-router-dom";

import { selectItem } from "./actions/actions";

import './App.css';

import SideNav from "./components/SideNav"
import ContentItem from "./components/ContentItem"
import { Provider } from "react-redux";
import store from "./store";
import Header from "./components/Header";

function App() {
    const movementScaler = 10;
    document.onmousemove = handleMouseMove;

    function handleMouseMove(event) {
        let target = document.getElementById("moveTarget")
        const height = window.innerHeight;
        const width = window.innerWidth;
        if (width > 1300) {
            let yAxisDegree = event.pageX / width * 40 - 20;
            let xAxisDegree = event.pageY / height * -1 * 40 + 20;
            yAxisDegree = yAxisDegree/movementScaler;
            xAxisDegree = xAxisDegree/movementScaler;
            target.style.transform = `rotateY(${yAxisDegree}deg) rotateX(${xAxisDegree}deg)`;
            // Set the sheen position
            setSheenPosition(event.pageX / width, event.pageY / width);
        } else {
            target.style.transform = `rotateY(0deg) rotateX(0deg)`;
        }
    }

    function setSheenPosition(xRatio, yRatio) {
        // This creates a "distance" up to 400px each direction to offset the sheen
        const xOffset = 1 - (xRatio - 0.5) * 800;
        const yOffset = 1 - (yRatio - 0.5) * 800;
        let target = document.getElementById("moveTarget")
        target.style.setProperty('--sheenX', `${xOffset}px`)
        target.style.setProperty('--sheenY', `${yOffset}px`)
    }

    return (
        <Provider store={store}>
            <div className="app bg" style={{perspective: "800px"}}>
                <div className="main" id="moveTarget">
                    <SideNav></SideNav>
                    <div className="right-bit">
                        <Header></Header>
                        <ContentItem></ContentItem>
                    </div>
                </div>
            </div>
        </Provider>
    );
}

export default App;