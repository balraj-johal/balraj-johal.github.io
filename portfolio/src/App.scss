@import url('https://fonts.googleapis.com/css2?family=VT323&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Roboto+Mono:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;1,100;1,200;1,300;1,400;1,500;1,600&display=swap');
.App {
    text-align: center;
}

.App-logo {
    height: 40vmin;
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .App-logo {
        animation: App-logo-spin infinite 20s linear;
    }
}

.App-header {
    background-color: #282c34;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
}

.App-link {
    color: #61dafb;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}

.bg {
    position: absolute;
    top: 0;
    left: 0;
    height: 100vh;
    width: 100vw;
    // background-image: url("./res/sarahdarkslow.gif");
    background-image: url("./res/bgs/1.gif");
    // background-image: url("./res/bgs/2.gif");
    // background-image: url("./res/bgs/3.gif");
    background-size:100% 100%;
    background-color: #cccccc;
}
.app {
    display: flex;
    align-items: center;
    justify-content: center;
    .main {
        position: relative;
        height: 760px;
        width: 1250px;
        background-color: rgba(148,148,148, 0.34);
        color: white;
        overflow: hidden; /* Try removing this to see how the sheen works! */
        --sheenX: 0; /* Set these with JavaScript */
        --sheenY: 0;

        .side-nav {
            width: 335px;
            height: 100%;
            overflow-y: scroll;
            font-family: "VT323";
            .list-header {
                padding-left: 20px;
                font-size: 30px;
                font-weight: 500;
                margin: 22px auto;
            }
            ul {
                list-style-type: none;
                margin: 0;
                margin-top: -10px;
                padding: 0;
                li {
                    padding: 10px 15px 11px 32px;
                    font-size: 22px;
                }
                .highlight {
                    background-color: rgba(255, 255, 255, 0.25);
                }
            }
        }
        .right-bit {
            position: absolute;
            top: 0;
            left: 335px;
            height: 100%;
            width: 915px;

            .header {
                display: flex;
                align-items: center;
                justify-content: space-between;
                padding-left: 26px;
                height: 70px;
                font-family: "VT323";

                h1 {
                    font-size: 42px;
                    font-weight: bold;
                    width: 100%;
                    padding-right: 26px;
                    text-align: right;
                    animation-fill-mode: forwards;
                }
                span {
                    font-weight: normal;
                    font-size: 30px;
                    margin-right: 5px;
                    position: absolute;
                    top: 20px;
                    right: 15px;
                    animation-fill-mode: forwards;
                }
            }
            .content-item {
                height: 690px;
                overflow-y: scroll;

                .image-container {
                    height: 530px;
                    img {
                        height: 100%;
                        width: 100%;
                        object-fit: cover;
                    }
                }
                .content {
                    font-family: 'Roboto Mono', monospace;
                    padding: 2em 2em;
                }
            }
        }
    }
}

.main::after {
    content: "";
    position: absolute;
    top: -400px;
    right: -400px;
    bottom: -400px;
    left: -400px;
    background: linear-gradient(217deg, rgba(255,255,255,0), rgba(255,255,255,0) 35%, rgba(255, 255, 255, 0.034) 45%, rgba(255, 255, 255, 0.089) 50%, rgba(255,255,255,0) 60%, rgba(255,255,255,0) 100%);
    transform: translateX(var(--sheenX)) translateY(var(--sheenY));
    pointer-events: none;
}
.hideit {
    display: none;
}

.moveitleft {
    animation: moveleft 1.2s ease-in-out;
}
.moveitright {
    animation: moveright 1.2s ease-in-out;
}
.movetitleleft {
    animation: movelefttitle 1.2s ease-in-out;
}
.movetitleright {
    animation: moverighttitle 1.2s ease-in-out;
}
@keyframes moveleft {
    0% {
        padding-right: 26px;
    }
    100% {
        padding-right: 77%;
    }
}
@keyframes movelefttitle {
    0% {
        margin-right: -77%;
    }
    100% {
        padding-right: 5px;
    }
}
@keyframes moveright {
    0% {
        padding-right: 77%;
    }
    100% {
        padding-right: 26px;
    }
}
@keyframes moverighttitle {
    0% {
        padding-right: 5px;
    }
    100% {
        margin-right: -77%;
    }
}