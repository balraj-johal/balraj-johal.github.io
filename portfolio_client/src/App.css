@import url("https://fonts.googleapis.com/css2?family=VT323&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Roboto+Mono:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;1,100;1,200;1,300;1,400;1,500;1,600&display=swap");
@media (max-width: 1300px) {
  .app {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
    -webkit-box-pack: center;
        -ms-flex-pack: center;
            justify-content: center;
  }
  .app .main {
    height: 760px;
    width: 1250px;
  }
}

@media (max-width: 576px) {
  .app {
    overflow: hidden;
  }
  .bg::before {
    -webkit-filter: blur(4px);
            filter: blur(4px);
    -webkit-transform: scale(1.1);
            transform: scale(1.1);
  }
  .mobile-wrap {
    width: 100%;
  }
  .header-selected-item {
    display: none;
  }
  .header h1, .mobile-nav {
    font-family: "VT323";
  }
  .carousel-wrap {
    scrollbar-width: none;
    -ms-overflow-style: none;
    /* IE and Edge */
    display: -ms-grid;
    display: grid;
    will-change: transform;
    -ms-flex-line-pack: center;
        align-content: center;
    overflow-x: auto;
    overflow-y: hidden;
    -ms-scroll-snap-coordinate: 0 0;
        scroll-snap-coordinate: 0 0;
    -ms-scroll-snap-points-x: repeat(100%);
        scroll-snap-points-x: repeat(100%);
    -ms-scroll-snap-type: x mandatory;
        scroll-snap-type: x mandatory;
    -webkit-overflow-scrolling: touch;
  }
  .carousel-wrap::-webkit-scrollbar {
    display: none;
  }
  .carousel-wrap .mobile-content-item {
    height: 80vh;
    scroll-snap-align: start;
    width: 100vw;
    font-family: "Roboto Mono";
    font-weight: normal;
  }
  .carousel-wrap .mobile-content-item .image-container {
    height: 30vh;
  }
  .carousel-wrap .mobile-content-item .image-container img {
    height: 100%;
    width: 100%;
    -o-object-fit: cover;
       object-fit: cover;
  }
  .carousel-wrap .mobile-content-item .content {
    padding: 5px 15px 15px 15px;
    font-size: 13px;
    height: 42vh;
    overflow-y: auto;
  }
  .carousel-wrap .mobile-content-item .content h1 {
    font-size: 18px;
  }
  .carousel-wrap .mobile-content-item .swipe-prompt {
    width: 90vw;
    text-align: right;
    margin: 3vh 5vw;
  }
}

@media (min-width: 576px) {
  .moveitleft, .moveitright {
    position: absolute;
    left: 0;
    width: 25% !important;
  }
}

@media (min-width: 768px) and (max-width: 1300px) {
  .header {
    margin-top: 2px;
    position: relative;
    height: 64px;
  }
  .header h1 {
    font-size: 31px !important;
    -webkit-animation-fill-mode: forwards;
            animation-fill-mode: forwards;
    -webkit-transition: 1s ease;
    transition: 1s ease;
  }
  .header div {
    width: 100%;
    text-align: right;
  }
  .moveitleft {
    -webkit-animation: moveleft 1.2s ease-in-out !important;
            animation: moveleft 1.2s ease-in-out !important;
    padding: 0 0 0 15px;
  }
  .moveitright {
    -webkit-animation: moveright 1.2s ease-in-out !important;
            animation: moveright 1.2s ease-in-out !important;
    padding: 0 15px 0 0;
  }
  .movetitleleft {
    -webkit-animation: movelefttitle 1.2s ease-in-out;
            animation: movelefttitle 1.2s ease-in-out;
  }
  .movetitleright {
    -webkit-animation: moverighttitle 1.2s ease-in-out;
            animation: moverighttitle 1.2s ease-in-out;
  }
  .side-nav .list-header {
    font-size: 26px;
  }
  .side-nav li {
    font-size: 21px;
  }
  .right-bit .header div {
    padding-right: 10px;
    left: 0;
    font-size: 26px !important;
  }
  .right-bit .content-item {
    height: 93vh;
  }
}

@media (min-width: 1300px) {
  .app {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
    -webkit-box-pack: center;
        -ms-flex-pack: center;
            justify-content: center;
  }
  .app .main {
    height: 770px !important;
    width: 1250px !important;
  }
  .app .main .side-nav {
    width: 335px !important;
  }
  .app .main .side-nav .list-header {
    font-size: 30px;
  }
  .app .main .side-nav .list-item {
    font-size: 22px;
  }
  .app .main .right-bit {
    position: absolute;
    top: 0;
    left: 335px;
    height: 100%;
    width: 915px !important;
  }
  .app .main .right-bit .header {
    height: 70px;
  }
  .app .main .right-bit .header div {
    position: absolute;
    top: 20px;
    right: 15px;
  }
  .app .main .right-bit .content-item {
    height: 690px;
    overflow-y: scroll;
  }
  .moveitleft {
    -webkit-animation: moveleft 1.2s ease-in-out;
            animation: moveleft 1.2s ease-in-out;
  }
  .moveitright {
    -webkit-animation: moveright 1.2s ease-in-out;
            animation: moveright 1.2s ease-in-out;
  }
  .movetitleleft {
    -webkit-animation: movelefttitle 1.2s ease-in-out;
            animation: movelefttitle 1.2s ease-in-out;
  }
  .movetitleright {
    -webkit-animation: moverighttitle 1.2s ease-in-out;
            animation: moverighttitle 1.2s ease-in-out;
  }
}

.bg {
  position: absolute;
  top: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
}

.bg::before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  background-image: url("./res/bgs/1.gif");
  background-size: 100% 100%;
  background-color: red;
}

.app {
  -webkit-perspective: 800px;
          perspective: 800px;
}

.app .main {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 100%;
  width: 100%;
  background-color: rgba(148, 148, 148, 0.34);
  color: white;
  overflow: hidden;
  --sheenX: 0;
  --sheenY: 0;
}

.app .main .side-nav {
  user-select: none;
  /* supported by Chrome and Opera */
  -webkit-user-select: none;
  /* Safari */
  -khtml-user-select: none;
  /* Konqueror HTML */
  -moz-user-select: none;
  /* Firefox */
  -ms-user-select: none;
  /* Internet Explorer/Edge */
  width: 30%;
  height: 100%;
  overflow-y: scroll;
  font-family: "VT323";
}

.app .main .side-nav .list-header {
  padding-left: 20px;
  font-weight: 500;
  margin: 22px auto;
}

.app .main .side-nav ul {
  list-style-type: none;
  margin: 0;
  margin-top: -10px;
  padding: 0;
}

.app .main .side-nav ul li {
  padding: 10px 15px 11px 32px;
}

.app .main .side-nav ul .highlight {
  background-color: rgba(255, 255, 255, 0.25);
}

.app .main .right-bit {
  width: 70%;
}

.app .main .right-bit .header {
  user-select: none;
  /* supported by Chrome and Opera */
  -webkit-user-select: none;
  /* Safari */
  -khtml-user-select: none;
  /* Konqueror HTML */
  -moz-user-select: none;
  /* Firefox */
  -ms-user-select: none;
  /* Internet Explorer/Edge */
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  padding-left: 26px;
  font-family: "VT323";
}

.app .main .right-bit .header h1 {
  font-size: 42px;
  font-weight: bold;
  -webkit-animation-fill-mode: forwards !important;
          animation-fill-mode: forwards !important;
}

.app .main .right-bit .header div {
  font-weight: normal;
  font-size: 30px;
  margin-right: 5px;
  -webkit-animation-fill-mode: forwards;
          animation-fill-mode: forwards;
  -webkit-transition: 1s ease;
  transition: 1s ease;
}

.app .main .right-bit .content-item {
  overflow-y: scroll;
}

.app .main .right-bit .content-item .image-container {
  height: 530px;
}

.app .main .right-bit .content-item .image-container img {
  height: 100%;
  width: 100%;
  -o-object-fit: cover;
     object-fit: cover;
}

.app .main .right-bit .content-item .content {
  font-family: 'Roboto Mono', monospace;
  padding: 2em 2em;
}

.mobile-nav ul {
  list-style-type: none;
}

.mobile-nav ul li {
  float: left;
  padding: 10px 10px;
}

.mobile-nav #category-list {
  scrollbar-width: none;
  -ms-overflow-style: none;
  /* IE and Edge */
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  width: 400px;
  overflow-x: auto;
}

.mobile-nav #category-list::-webkit-scrollbar {
  display: none;
}

.mobile-nav #category-list .list-item {
  padding: 10px 20px;
  width: auto;
  height: 10px;
}

.mobile-nav #category-list .list-item a {
  white-space: nowrap;
}

.main::after {
  content: "";
  position: absolute;
  top: -400px;
  right: -400px;
  bottom: -400px;
  left: -400px;
  background: linear-gradient(217deg, rgba(255, 255, 255, 0), rgba(255, 255, 255, 0) 35%, rgba(255, 255, 255, 0.034) 45%, rgba(255, 255, 255, 0.089) 50%, rgba(255, 255, 255, 0) 60%, rgba(255, 255, 255, 0) 100%);
  -webkit-transform: translateX(var(--sheenX)) translateY(var(--sheenY));
          transform: translateX(var(--sheenX)) translateY(var(--sheenY));
  pointer-events: none;
}

.hideit {
  display: none;
}

@-webkit-keyframes moveleft {
  0% {
    right: 0%;
    left: 73%;
    text-align: right;
  }
  100% {
    right: 100%;
    left: 0%;
    text-align: left;
  }
}

@keyframes moveleft {
  0% {
    right: 0%;
    left: 73%;
    text-align: right;
  }
  100% {
    right: 100%;
    left: 0%;
    text-align: left;
  }
}

@-webkit-keyframes moveright {
  0% {
    right: 100%;
    left: 0%;
    text-align: left;
  }
  100% {
    right: 0%;
    left: 73%;
    text-align: right;
  }
}

@keyframes moveright {
  0% {
    right: 100%;
    left: 0%;
    text-align: left;
  }
  100% {
    right: 0%;
    left: 73%;
    text-align: right;
  }
}

@-webkit-keyframes movelefttitle {
  0% {
    color: rgba(255, 255, 255, 0);
  }
  100% {
    color: white;
  }
}

@keyframes movelefttitle {
  0% {
    color: rgba(255, 255, 255, 0);
  }
  100% {
    color: white;
  }
}

@-webkit-keyframes moverighttitle {
  100% {
    color: white;
  }
  0% {
    color: rgba(255, 255, 255, 0);
  }
}

@keyframes moverighttitle {
  100% {
    color: white;
  }
  0% {
    color: rgba(255, 255, 255, 0);
  }
}

@-webkit-keyframes movelefta {
  0% {
    left: 50vw;
  }
  100% {
    left: 15px;
  }
}

@keyframes movelefta {
  0% {
    left: 50vw;
  }
  100% {
    left: 15px;
  }
}

@-webkit-keyframes moverighta {
  0% {
    left: 15px;
  }
  100% {
    left: 50vw;
  }
}

@keyframes moverighta {
  0% {
    left: 15px;
  }
  100% {
    left: 50vw;
  }
}

#content-scrollbar-cust, #sidenav-scrollbar-cust, .cust-scroll {
  scrollbar-color: rgba(199, 199, 199, 0.5) rgba(1, 0, 0, 0);
  scrollbar-width: thin;
}

#content-scrollbar-cust::-webkit-scrollbar-track,
#sidenav-scrollbar-cust::-webkit-scrollbar-track,
.cust-scroll::-webkit-scrollbar-track {
  border-radius: 10px;
  background-color: #f5f5f500;
}

#content-scrollbar-cust::-webkit-scrollbar,
#sidenav-scrollbar-cust::-webkit-scrollbar,
.cust-scroll::-webkit-scrollbar {
  width: 8px;
  background-color: #f5f5f500;
}

#content-scrollbar-cust::-webkit-scrollbar-thumb,
#sidenav-scrollbar-cust::-webkit-scrollbar-thumb,
.cust-scroll::-webkit-scrollbar-thumb {
  border-radius: 10px;
  background-color: rgba(173, 173, 173, 0.5);
}
/*# sourceMappingURL=App.css.map */