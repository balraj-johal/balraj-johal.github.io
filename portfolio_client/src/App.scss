// @use "styles/mobile";

@import url('https://fonts.googleapis.com/css2?family=VT323&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Roboto+Mono:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;1,100;1,200;1,300;1,400;1,500;1,600&display=swap');

@mixin noDragText {
    user-select: none; /* supported by Chrome and Opera */
   -webkit-user-select: none; /* Safari */
   -khtml-user-select: none; /* Konqueror HTML */
   -moz-user-select: none; /* Firefox */
   -ms-user-select: none; /* Internet Explorer/Edge */
}

$largest: 1300px;
@media (max-width: $largest) {
    .app {
        display: flex;
        align-items: center;
        justify-content: center;
        .main {
            height: 760px;
            width: 1250px;
        }
    }
}

@media (min-width: 576px) {
    .moveitleft, .moveitright {
        position: absolute;
        left: 0;
        width: 25% !important;
    }
}
@media (min-width: 768px) and (max-width: $largest) {
    .header {
        margin-top: 2px;
        position: relative;
        height: 64px;
        h1 {
            font-size: 31px !important;
            animation-fill-mode: forwards;
            transition: 1s ease;
        }
        div {
            width: 100%;
            text-align: right;
        }
    }
    .moveitleft {
        animation: moveleft 1.2s ease-in-out !important;
        padding: 0 0 0 15px;
    }
    .moveitright {
        animation: moveright 1.2s ease-in-out !important;
        padding: 0 15px 0 0;
    }
    .movetitleleft {
        animation: movelefttitle 1.2s ease-in-out;
    }
    .movetitleright {
        animation: moverighttitle 1.2s ease-in-out;
    }
    .side-nav {
        .list-header {
            font-size: 26px;
        }
        li {
            font-size: 21px;
        }
    }
    .right-bit {
        .header {
            div {
                padding-right: 10px;
                left: 0;
                font-size: 26px !important;
            }
        }
        .content-item {
            height: 93vh;
        }
    }
}

@media (min-width: $largest) {
    .moveitleft {
        animation: moveleft 1.2s ease-in-out;
    }
    .moveitright {
        animation: moveright 1.2s ease-in-out;
    }
    .movetitleleft {
        animation: movelefttitle 1.2s ease-in-out;
    }
    .movetitleright {
        animation: moverighttitle 1.2s ease-in-out;
    }
    .app {
        display: flex;
        align-items: center;
        justify-content: center;
        .main {
            height: 770px !important;
            width: 1250px !important;
            
            .side-nav {
                width: 335px !important;
                .list-header {
                    font-size: 30px;
                }
                .list-item {
                    font-size: 22px;
                }
            }
            .right-bit {
                .header {
                    height: 70px;
                }
                position: absolute;
                top: 0;
                left: 335px;
                height: 100%;
                width: 915px !important;
                .header {
                    div {
                        position: absolute;
                        top: 20px;
                        right: 15px;
                    }
                }
                .content-item {
                    height: 690px;
                    overflow-y: scroll;
                }
            }
        }
    }
}


.bg {
    position: absolute;
    top: 0;
    left: 0;
    height: 100vh;
    width: 100vw;
    // background-image: url("./res/sarahdarkslow.gif");
    background-image: url("./res/bgs/1.gif");
    // background-image: url("./res/bgs/2.gif");
    // background-image: url("./res/bgs/3.gif");
    background-size:100% 100%;
    background-color: #cccccc;
}
.app {
    .main {
        display: flex;
        position: relative;
        height: 100%;
        width: 100%;
        background-color: rgba(148,148,148, 0.34);
        color: white;
        overflow: hidden;
        --sheenX: 0;
        --sheenY: 0;
        .side-nav {
            @include noDragText();
            width: 30%;
            height: 100%;
            overflow-y: scroll;
            font-family: "VT323";
            .list-header {
                padding-left: 20px;
                font-weight: 500;
                margin: 22px auto;
            }
            ul {
                list-style-type: none;
                margin: 0;
                margin-top: -10px;
                padding: 0;
                li {
                    padding: 10px 15px 11px 32px;
                }
                .highlight {
                    background-color: rgba(255, 255, 255, 0.25);
                }
            }
        }
        .right-bit {
            width: 70%;
            .header {
                @include noDragText();
                display: flex;
                align-items: center;
                justify-content: space-between;
                padding-left: 26px;
                font-family: "VT323";
                h1 {
                    font-size: 42px;
                    font-weight: bold;
                    animation-fill-mode: forwards !important;
                }
                div {
                    font-weight: normal;
                    font-size: 30px;
                    margin-right: 5px;
                    animation-fill-mode: forwards;
                    transition: 1s ease;
                }
            }
            .content-item {
                overflow-y: scroll; 
                .image-container {
                    height: 530px;
                    img {
                        height: 100%;
                        width: 100%;
                        object-fit: cover;
                    }
                }
                .content {
                    font-family: 'Roboto Mono', monospace;
                    padding: 2em 2em;
                }
            }
        }
    }
}

@mixin hideScrollbar {
    scrollbar-width: none;
    -ms-overflow-style: none;  /* IE and Edge */
    &::-webkit-scrollbar {
        display: none;
    }
}
.mobile-nav {
    ul {
        list-style-type: none;
        li {
            float: left;
            padding: 10px 10px;
        }
    }
    #category-list {
        @include hideScrollbar();
        display: flex;
        width: 400px;
        overflow-x: auto;
        .list-item {
            padding: 10px 20px;
            width: auto;
            height: 10px;
            // display: block ruby;
            a {
                white-space: nowrap;
            }
        }
    }
}

.main::after {
    content: "";
    position: absolute;
    top: -400px;
    right: -400px;
    bottom: -400px;
    left: -400px;
    background: linear-gradient(217deg, rgba(255,255,255,0), rgba(255,255,255,0) 35%, rgba(255, 255, 255, 0.034) 45%, rgba(255, 255, 255, 0.089) 50%, rgba(255,255,255,0) 60%, rgba(255,255,255,0) 100%);
    transform: translateX(var(--sheenX)) translateY(var(--sheenY));
    pointer-events: none;
}
.hideit {
    display: none;
}



@keyframes moveleft {
    0% {
        right: 0%;
        left: 73%;
        text-align: right;
    }
    100% {
        right: 100%;
        left: 0%;
        text-align: left;
    }
}
@keyframes moveright {
    0% {
        right: 100%;
        left: 0%;
        text-align: left;
    }
    100% {
        right: 0%;
        left: 73%;
        text-align: right;
    }
}
@keyframes movelefttitle {
    0% {
        color: rgba(255,255,255,0);
    }
    100% {
        color: white;
    }
}
@keyframes moverighttitle {
    100% {
        color: white;
    }
    0% {
        color: rgba(255,255,255,0);
    }
}
@keyframes movelefta {
    0% {
        left: 50vw;
    }
    100% {
        left: 15px;
    }
}
@keyframes moverighta {
    0% {
        left: 15px;
    }
    100% {
        left: 50vw;
    }
}

//SCROLLBAR

#content-scrollbar-cust, #sidenav-scrollbar-cust {
    scrollbar-color: gray rgba(1,0,0,0);
    scrollbar-width: thin;
}
#content-scrollbar-cust::-webkit-scrollbar-track
{
	border-radius: 10px;
	background-color: #f5f5f500;
}
#content-scrollbar-cust::-webkit-scrollbar
{
	width: 8px;
	background-color: #f5f5f500;
}
#content-scrollbar-cust::-webkit-scrollbar-thumb
{
	border-radius: 10px;
	background-color: rgb(173, 173, 173);
}

#sidenav-scrollbar-cust::-webkit-scrollbar-track
{
	border-radius: 10px;
	background-color: #f5f5f500;
}
#sidenav-scrollbar-cust::-webkit-scrollbar
{
	width: 8px;
	background-color: #f5f5f500;
}
#sidenav-scrollbar-cust::-webkit-scrollbar-thumb
{
	border-radius: 10px;
	background-color: rgb(173, 173, 173);
}