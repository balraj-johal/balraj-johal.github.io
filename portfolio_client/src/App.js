import './App.css';

import { Provider } from "react-redux";
import store from "./store";
import { useEffect, useState } from 'react';

import TopNav from "./components/TopNav"
import SideNav from "./components/SideNav"
import ContentItem from "./components/ContentItem"
import ContentCarousel from "./components/ContentCarousel"
import Header from "./components/Header";

function App() {

    //check if we should render mobile version
    let [mobile, setMobile] = useState(false);
    let isMobile = () => {
        if (window.innerWidth <= 576) {
            setMobile(true);
        } else {
            setMobile(false);
        }
    }
    useEffect(() => {
        window.addEventListener('resize', isMobile);
        if (window.innerWidth <= 576) {
            setMobile(true);
        } else {
            setMobile(false);
        }
        
        return () => {
            window.removeEventListener('resize', isMobile);
        };
    })
    
    const MOVEMENT_SCALE = 10;
    document.onmousemove = handleMouseMove;
    //3D window effect
    function handleMouseMove(event) {
        let target = document.getElementById("moveTarget")
        const height = window.innerHeight;
        const width = window.innerWidth;
        if (width > 1300) {
            let yAxisDegree = event.pageX / width * 40 - 20;
            let xAxisDegree = event.pageY / height * -1 * 40 + 20;
            yAxisDegree = yAxisDegree/MOVEMENT_SCALE;
            xAxisDegree = xAxisDegree/MOVEMENT_SCALE;
            target.style.transform = `rotateY(${yAxisDegree}deg) rotateX(${xAxisDegree}deg)`;
            // Set the sheen position
            setSheenPosition(event.pageX / width, event.pageY / width);
        } else {
            target.style.transform = `rotateY(0deg) rotateX(0deg)`;
        }
    }
    function setSheenPosition(xRatio, yRatio) {
        // This creates a "distance" up to 400px each direction to offset the sheen
        const xOffset = 1 - (xRatio - 0.5) * 800;
        const yOffset = 1 - (yRatio - 0.5) * 800;
        let target = document.getElementById("moveTarget")
        target.style.setProperty('--sheenX', `${xOffset}px`)
        target.style.setProperty('--sheenY', `${yOffset}px`)
    }

    return (
        <Provider store={store}>
            <div className="app bg">
                <div className="main" id="moveTarget">
                    {(mobile) ? null : <SideNav mobile={mobile}></SideNav>}
                    <div className={`${(mobile) ? "mobile-wrap" : "right-bit" }`}>
                        <div className="top-wrap">
                            <Header></Header>
                            {(mobile) ?  <TopNav /> : null}
                        </div>
                        {(mobile) ? <ContentCarousel /> : <ContentItem />}
                    </div>
                </div>
            </div>
        </Provider>
    );
}

export default App;