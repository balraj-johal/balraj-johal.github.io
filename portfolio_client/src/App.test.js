import { render, screen, cleanup } from '@testing-library/react';
import App from './App';

afterEach(() => {
    cleanup();
})

test('renders app root div', () => {
    render(<App />);
    const appRootElement = screen.getByTestId("app-root");
    expect(appRootElement).toBeInTheDocument();
});

it('should take a snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
})