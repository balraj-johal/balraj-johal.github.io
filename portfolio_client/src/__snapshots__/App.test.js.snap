// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="app bg"
    data-testid="app-root"
  >
    <div
      class="main"
      id="transformTarget"
    >
      <div
        class="side-nav"
        id="sidenav-scrollbar-cust"
      >
        <h2
          class="list-header"
        >
          Professional Work
        </h2>
        <ul>
          <div>
            <li
              class="list-item"
              id="list-item-0"
            >
              Routes
            </li>
          </div>
          <div>
            <li
              class="list-item"
              id="list-item-1"
            >
              XR Stories - Treo
            </li>
          </div>
          <div>
            <li
              class="list-item"
              id="list-item-2"
            >
              IoC Student Enterprises Website
            </li>
          </div>
          <div>
            <li
              class="list-item"
              id="list-item-3"
            >
              Robot Theatre
            </li>
          </div>
          <div>
            <li
              class="list-item"
              id="list-item-4"
            >
              AR/AI Visual Programming Toolkit
            </li>
          </div>
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </ul>
        <h2
          class="list-header"
        >
          Other Stuff
        </h2>
        <ul>
          <div />
          <div />
          <div />
          <div />
          <div />
          <div>
            <li
              class="list-item"
              id="list-item-5"
            >
              TensorFlow Biomedical Image Segmentation
            </li>
          </div>
          <div>
            <li
              class="list-item"
              id="list-item-6"
            >
              This website [WIP]
            </li>
          </div>
          <div>
            <li
              class="list-item"
              id="list-item-7"
            >
              Paradoujin.com - Underground Music Blog
            </li>
          </div>
          <div>
            <li
              class="list-item"
              id="list-item-8"
            >
              Constellations
            </li>
          </div>
          <div>
            <li
              class="list-item"
              id="list-item-9"
            >
              I just need some ******* eggs [WIP]
            </li>
          </div>
          <div />
          <div />
          <div />
        </ul>
      </div>
      <div
        class="right-bit"
      >
        <div
          class="top-wrap"
        >
          <div
            class="header"
          >
            <h1
              class="moveitright"
            >
              BALRAJ JOHAL
            </h1>
            <div
              class="header-selected-item 
                    transparent 
                    movetitleright
                "
            />
          </div>
        </div>
        <div
          class="content-item"
          id="content-scrollbar-cust"
        >
          <div
            class="hideit image-container"
          >
            <img
              draggable="false"
              src="/res/"
            />
          </div>
          <div
            class="content"
          >
            
        
            <p>
              
            Hey I'm Balraj! I'm a Software Developer with
            experience working on full-stack projects with a range of 
            partners across varied disciplines.
        
            </p>
            <p>
              
            I'm passionate about creative solutions to problems and
            especially creative applications of digital technology.
        
            </p>
            

        
            <div
              class="about-img-wrap"
            >
              
            
              <img
                draggable="false"
                src="/res/me.jpg"
              />
              
            
            
              <span
                class="it-me"
              >
                it me
              </span>
              
        
            </div>
            

        
            <a
              href="/res/CV - Balraj Johal.pdf"
            >
              
            
              <div
                class="cv-link"
              >
                
                MY CV
            
              </div>
              
        
            </a>
            

        
            <p>
              
            Get in touch with me if you have somthing cool for us to work on!
        
            </p>
            
        
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
